src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Resolve.java

```diff
@@ -3834,7 +3834,7 @@ Symbol findSelfContaining(DiagnosticPosition pos,
      */
     Symbol findLocalClassOwner(Env<AttrContext> env, TypeSymbol c) {
         Symbol owner = c.owner;
-        Assert.check(owner.kind == MTH);
+        Assert.check(owner.kind == MTH || owner.kind == VAR);
```

Comment:
We now need to generalize this, so that it works for both method and variable owners.